src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrayOfEntries = sourceString.split(';');
+  const arrayOfKeysAndValues = arrayOfEntries.map((entry) => entry.split(':'));
+  const trimmedArray = arrayOfKeysAndValues.map(function (item) {
+    if (item.length === 2) {
+      return [item[0].trim(), item[1].trim()];
+    }
+
+    return item;
+  });
+  const filtredArray = trimmedArray.filter((entry) => entry.length > 1);
```

Comment:
The filter condition `entry.length > 1` is correct, but ensure that the input string is well-formed. If any entry does not contain a colon, it will be filtered out, which might not be the intended behavior. Consider adding error handling or logging for such cases.